app/hotels/src/map/allHotels/NewAllHotelsMap.js

```diff
@@ -0,0 +1,110 @@
+// @flow strict
+
+import * as React from 'react';
+import { View } from 'react-native';
+import { graphql, PublicApiRenderer } from '@kiwicom/mobile-relay';
+import { DateFormatter } from '@kiwicom/mobile-localization';
+import { StyleSheet, AdaptableLayout } from '@kiwicom/mobile-shared';
+import type { NavigationType } from '@kiwicom/mobile-navigation';
+
+import FilterStripe from '../../filter/FilterStripe';
+import {
+  withHotelsContext,
+  type RoomConfigurationType,
+} from '../../HotelsContext';
+import HotelsFilterContext from '../../HotelsFilterContext';
+import MapScreen from './MapScreen';
+import { sanitizeHotelFacilities } from '../../GraphQLSanitizers';
+import type { NewAllHotelsMapQueryResponse } from './__generated__/NewAllHotelsMapQuery.graphql';
+
+type Props = {|
+  +cityId: string,
+  +checkin: Date | null,
+  +checkout: Date | null,
+  +currency: string,
+  +roomsConfiguration: RoomConfigurationType,
+  +navigation: NavigationType,
+|};
+
+class NewAllHotelsMap extends React.Component<Props> {
+  openSingleHotel = (hotelId: string) => {
+    this.props.navigation.navigate('SingleHotel', {
+      hotelId,
+      checkin: this.props.checkin,
+      checkout: this.props.checkout,
+      roomsConfiguration: this.props.roomsConfiguration,
+    });
+  };
+
+  renderInnerComponent = (props: NewAllHotelsMapQueryResponse) => (
+    <MapScreen
+      data={props.allAvailableHotels}
+      onOpenSingleHotel={this.openSingleHotel}
+    />
+  );
+
+  render = () => {
+    const {
+      checkin,
+      checkout,
+      cityId,
+      currency,
+      roomsConfiguration,
+    } = this.props;
+    if (checkin === null || checkout === null) {
```

Comment:
TODO warning too?

Reply:
Yes, should be

Reply:
Actually, here it is not needed. The first screen the user will see is AllHotelsScreen which will show error. 

On tablet the user will se the error on the left half. I think that should be sufficient. Or do you think otherwise?

Reply:
Then if the map is not 'functioning', the button to access it should be disabled if there is an error on the first screen. Or if the button is not disabled, I would still show an error for people not to think it is a bug that the map is not loading.

Reply:
I guess we could disable the map button when I fix the header.